cdap-cli/src/main/java/co/cask/cdap/cli/command/security/AddRoleToPrincipalCommand.java

```diff
@@ -47,13 +47,12 @@ public void perform(Arguments arguments, PrintStream output) throws Exception {
     String principalName = arguments.get("principal-name");
     client.addRoleToPrincipal(new Role(roleName), new Principal(principalName, Principal.PrincipalType.valueOf
       (principalType.toUpperCase())));
-    output.printf("Successfully added role '%s' to %s %s\n", roleName, principalType, principalName);
+    output.printf("Successfully added role '%s' to '%s' '%s'\n", roleName, principalType, principalName);
   }
 
   @Override
   public String getPattern() {
-    return String.format("add role <%s> to %s <%s> %s <%s>", ArgumentName.ROLE_NAME,
-                         ArgumentName.PRINCIPAL_TYPE, ArgumentName.PRINCIPAL_TYPE, ArgumentName.PRINCIPAL_NAME,
+    return String.format("add role <%s> to <%s> <%s>", ArgumentName.ROLE_NAME, ArgumentName.PRINCIPAL_TYPE,
```

Comment:
but ``listPrivileges`` is a totally different operation from ``addRoleToPrincipal``. We're specifically talking about ``addRoleToPrincipal``, and for that operation, this check should not be done in every single extension. Adding a role to another role should not be supported by any extension unless we have a concrete use case for it.